tests/test_algorithm.py

```diff
@@ -313,6 +314,47 @@ def handle_data(context, data):
         with self.assertRaises(TypeError):
             algo.run()
 
+    @parameterized.expand([
+        (-1000, 'invalid_base'),
+        (0, 'invalid_base'),
+    ])
+    def test_invalid_capital_base(self, cap_base, name):
+        """
+        Test that the appropriate error is being raised and orders aren't
+        filled for algos with capital base <= 0
+        """
+        algo_text = """
+def initialize(context):
+    pass
+
+def handle_data(context, data):
+    order(sid(24), 1000)
+        """
+
+        sim_params = SimulationParameters(
+            start_session=pd.Timestamp("2006-01-03", tz='UTC'),
+            end_session=pd.Timestamp("2006-01-06", tz='UTC'),
+            capital_base=cap_base,
+            data_frequency="minute",
+            trading_calendar=self.trading_calendar
+        )
+
+        with self.assertRaises(ZeroCapitalError) as exc:
+            # make_algo will trace to TradingAlgorithm,
+            # where the exception will be raised
+            algo = self.make_algo(script=algo_text, sim_params=sim_params)
+        # It shouldn't be able to run the algorithm
```

Comment:
Comment indenting got weird here. When you say "it shouldn't" do you mean that you shouldn't be able to get to this line of code? Or that the return into `output` should be `None`?

Reply:
Changed the indentation and made the comment more clear. What I meant was that since self.make_algo() will raise the zero capital error, make_algo() may not finish executing, so algo.run() may then end up raising some errors

Reply:
So it looks like you changed the indentation of the comment, but the lines below it are still indented (thus they won't be run after that error is raised). So this test will still pass, no matter what assertions are made below and their correctness.

Reply:
I realized this yesterday. I scrapped the assertion for starting and ending capital base (to be equal) because I could no longer execute algo.run(). The only assertion I left is that the text of the error matches the ZeroCapitalError text. (I outdented the assertion this time)